src/main/java/millom/sandbox/mars/weather/resource/NasaResource.java

```diff
@@ -63,10 +59,73 @@ public Response getMarsWeather(@QueryParam("feed") String feed,
     }
   }
 
-  private Response getErrorResponse(String inputField) {
-    return Response.
-        status(Status.INTERNAL_SERVER_ERROR)
+  @Path("/earthdate/marsweather/")
+  @GET
+  public Response getMarsWeatherForEarthDate(
+      @QueryParam("feed") String feed,
+      @QueryParam("feedtype") String feedType,
+      @QueryParam("version") float version,
+      @QueryParam("category") String category,
+      @QueryParam("date") String date) {
+
+    Optional<Response> errorResponse = validate(feed, feedType, version, category);
+    if (errorResponse.isPresent()) {
+      return errorResponse.get();
+    }
+    if (date == null || date.length() == 0 || !date.matches("\\d{4}-\\d{2}-\\d{2}")) {
+      return getErrorResponse("date").get();
```

Comment:
getErrorResponse() method never returns null, so no need to return Optional from it.